rclcpp/test/rclcpp/executors/test_executors.cpp

```diff
@@ -525,10 +485,11 @@ TYPED_TEST(TestExecutorsStable, spinSome) {
   // count becomes too large, spin exits, or the 1 second timeout completes.
   auto start = std::chrono::steady_clock::now();
   while (
-    my_waitable->get_count() <= 1 &&
+    my_waitable->get_count() <= 10 &&
```

Comment:
why?

Reply:
If `spin_some(1s)` works as expected, the executor should execute only a single waitable event, and should ignore new notifications about the waitable being triggered more times.

If in the while loop (which triggers the waitable) we limit the count to 1, we are not testing the possibility of `spin_some` not working properly, that is, executing the waitable more than a single time. In order to test this situation, the waitable should have been triggered more than a single time (which currently could not happen if we limit the count to 1). So the test will be succesful:
 `EXPECT_EQ(1u, my_waitable->get_count())` even if `spin_some` didn't work as expected.

Now lets suppose `spin_some` doesn't work as expected, with the change `get_count() <= 10` , we can trigger the waitable many times (up to 10) and the executor might execute many waitable events. In this case, the test won't be succesful, if `my_waitable->get_count() = 10;`
`EXPECT_EQ(1u, my_waitable->get_count())` 

Furthermore, I think a `sleep(10ms)` should be added before `executor.spin_some(1s);`, to give time the executor to already have events when it starts spinning. Otherwise it might start spinning without having any events, so the spin exits, the test fails, but it was due it didn't have anything to do yet. What do you think?

Reply:
I have reverted this change also. What I said above is valid for the wait_set executors, which can execute as max one event per entity, per `spin_some` call.
It is still not here, but the `EventsExecutor` can process multiple events for the same entity in the same `spin_some` call, if at the time of calling it there were already many events ready to be executed for the same entity.

Reply:
> If in the while loop (which triggers the waitable) we limit the count to 1, we are not testing the possibility of spin_some not working properly, that is, executing the waitable more than a single time. In order to test this situation, the waitable should have been triggered more than a single time (which currently could not happen if we limit the count to 1). So the test will be succesful:

Good point, that seems to make sense to me.

> Furthermore, I think a sleep(10ms) should be added before executor.spin_some(1s);, to give time the executor to already have events when it starts spinning. Otherwise it might start spinning without having any events, so the spin exits, the test fails, but it was due it didn't have anything to do yet. What do you think?

The TestWaitable is always returning ready, so that doesn't seem to be a problem

> It is still not here, but the EventsExecutor can process multiple events for the same entity in the same spin_some call, if at the time of calling it there were already many events ready to be executed for the same entity.

That sounds reasonable to me